### PR TITLE
render components based on the annotations

### LINKED ISSUE #20837: [Declarative User Profile] Attributes tab doesn't dipslay attributes, values get lost and multi values get stored as single ones
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

user-profile

### Describe the bug

User attributes are no longer shown in the attributes tab.
image

They are only shown in the User details tab.
image

## Bugs

* When adding a user attribute at Attribute tab, other existing stored attribute values get lost
* When editing a multi value user attribute at the user details tab (e.g. modify "A,B" to "A,B,C", it is stored as single comma separated value "A,B,C")

### Version

21.1.1

### Expected behavior

- No data gets lost when adding attribute values
- Multi value attributes are not stored as single value when edited
- Attributes tab displays all stored attributes 

### Actual behavior

see section "Describe the bug" 

### How to Reproduce?

1. Define two attributes in the declarative user profile (in the screenshots above "tel" and "orgs" is defined).
2. Add "tel: 1111", "orgs: A" and "orgs:B" in the attributes tab
3. Modify "orgs: A,B" to "orgs: A,B,C" in user details tab will store it as "A,B,C"
4. Add attribute "tel: 1234" in attributes tab will delete the orgs value

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by edewit**
At the moment this only addresses the admin UI, but planned is to refactor this into a common components that work as well for the account ui

fixes: https://github.com/keycloak/keycloak/issues/20837

### CODE REVIEW

**Comment by edewit** [comment]
fixes: #20837

**Review by jonkoops — DISMISSED** [review]

**Review by jonkoops — DISMISSED** [review]

**Review by jonkoops — DISMISSED** [review]

File: js/apps/admin-ui/src/realm-settings/NewAttributeSettings.tsx [hunk 1]
```diff
@@ -151,7 +151,7 @@ export default function NewAttributeSettings() {
         "annotations",
         Object.entries(annotations || {}).map(([key, value]) => ({
           key,
-          value,
+          value: value as Record<string, unknown>,
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/NewAttributeSettings.tsx:154]: Looking at the type, it is the same as the one it is being cast to. So I think we can omit this cast.
```suggestion
          value,
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/realm-settings/NewAttributeSettings.tsx (hunk 1 above)
Comment (edewit) [reply js/apps/admin-ui/src/realm-settings/NewAttributeSettings.tsx:154]: the build failed when it was omitted

Comment (jonkoops) [reply js/apps/admin-ui/src/realm-settings/NewAttributeSettings.tsx:154]: Ahhh, I see. I didn't re-build the admin client, so I never got the new type.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**